### ISSUE TITLE
Update Search page to always show search input at the top and expense type tabs below

### ISSUE THREAD

**Issue by shawnborton** [Expensify/App#52317]
Right now we face some inconsistent behavior with the Search page depending on if you use a LHN nav item or if you search via the router. The LHN has default navigational items that don't use search input across the top of the page, but we do indeed show a search input if you search for something via the router. Furthermore, the default Search experience does not show a type selector across the top, however we are planning to show one across the top if you searched for something via the router. 

In order to clean up these consistencies, we're proposing these incremental updates to the search page:
- Always show a search input at the top of the page
- Do not show expense types in the LHN. Instead, use the LHN for default suggested searches as well as saved searches
    - The two default searches will be "All expenses" and "Expenses waiting on you" to start
- Move the status selector (All, drafts, outstanding, etc) to be within the RHP filters panel

That gives us something like this: 
![image](https://github.com/user-attachments/assets/8ed0575e-aabf-4c3d-9d87-f9bfc1b0a467)

https://github.com/user-attachments/assets/8661de75-afb9-40fb-99f1-f17ae490f615

On mobile, the idea is to use a UI like this where we use a full-page router experience that you can close in the top right corner after focusing: 
https://github.com/user-attachments/assets/26f03d60-1c4a-474d-9719-744779a4be93

On the note of filters, the idea is to move Status into the list of filters in the RHP:
![image](https://github.com/user-attachments/assets/05a306ec-43b7-4b9c-bb6c-cfd24b166544)

One small update though is we'd like to make it so you can select multiple statuses as once, which more closely matches OldDot's behavior for expense/report status selection: 
![image](https://github.com/user-attachments/assets/3679b0ea-e93e-4bc1-9cee-a009ef5dafd1)

cc @luacmartins @JmillsExpensify @trjExpensify @Expensify/design

**Comment by luacmartins** [comment]
@Kicu anyone from SWM interested in working on this one?

**Comment by Kicu** [comment]
@luacmartins hm, right now Im actually refactoring this code, so that the input has the autocomplete behavior similar to router, and so that the query always stays correct, if a user keeps editing it and resubmitting.

The change described here would be affecting the (react) components flow in the header of the Search Results page, and I think it would be better if someone doing this waits for my PR.
I can open a draft later today so you could add HOLD to this issue, and my PR should be done withing 1-2 days I think + review.

And yes, we would definitely like to work on this in SWM, I'll post this internally and someone will pick it up. For now perhaps assign me to this?

**Comment by Kicu** [comment]
here is the aforementioned PR: https://github.com/Expensify/App/pull/52568 still a draft, but now we can link it

**Comment by luacmartins** [comment]
I agree we should hold on your changes. Thanks for the linked PR and looking forward to getting someone from SWM assigned to the issue!

**Comment by shawnborton** [comment]
Another thing I wanted to note for this issue - I think it makes way more sense to bring the multi-select button out of the router so it sits below it like this:
![image](https://github.com/user-attachments/assets/731d9451-4e81-43aa-aeaa-d64fd3eb5128)

Curious what everyone thinks about that though.

**Comment by dannymcclain** [comment]
>I think it makes way more sense to bring the multi-select button out of the router so it sits below it like this

Where does it show up now? Inside the search input where the filters button is? Anyhoo, I agree and I think what you're showing in that mock looks good and makes sense. 👍

**Comment by shawnborton** [comment]
Yeah it currently replaces the filter icon within the search bar, which feels a little strange.

**Comment by dannymcclain** [comment]
Agree with that. Definitely like what you posted much better.

**Comment by shawnborton** [comment]
Not overdue, still working through this one.

**Comment by luacmartins** [comment]
Still holding

**Comment by Kicu** [comment]
Waiting on merge of: https://github.com/Expensify/App/pull/52568
I remember about this PR and someone from SWM will pick it up once 52568 is close to merging. Currently scope is growing a bit for the PR.

**Comment by trjExpensify** [comment]
The linked PR has merged, so shall we take this issue off hold? I'm also going to move this now into #migrate, so we track it there as a key foundational issue to get in place for the rest of our recently discussed plans. Excited for it!

**Comment by luacmartins** [comment]
We'll be working on a follow up to that PR to fully enable the features on the Search results page, so I don't think we should remove the hold quite yet

**Comment by trjExpensify** [comment]
Alright, sounds good!

**Comment by shawnborton** [comment]
Not overdue, we're working on it.

**Comment by SzymczakJ** [comment]
Hey! I’m Jakub Szymczak from Software Mansion, an expert agency, and I’d like to work on this issue!

**Comment by luacmartins** [comment]
https://github.com/Expensify/App/pull/53198 is merged. We can get started on this one.

**Comment by SzymczakJ** [comment]
Other tasks took me more, than I thought. I'll be starting working on this today.

**Comment by SzymczakJ** [comment]
> The two default searches will be "All expenses" and "Expenses waiting on you" to start

In case of default searches, this video confuses me(video in the issue description): 
<img width="821" alt="Screenshot 2024-12-11 at 17 15 52" src="https://github.com/user-attachments/assets/c124ca2f-c830-4a0a-bfad-82bf2f1e2e1c" />

Does it mean that "Expenses waiting on you" will be something, that by default gives you "outstanding" and "draft" status. Or will it be another API command, or something different. What should be the behaviour of default searches? Can we choose the "Expenses waiting on you" and then filter it even more with category filter(for example).

**Comment by dannymcclain** [comment]
Here's my take on this, @luacmartins @shawnborton @JmillsExpensify please feel free to correct me or add any additional thoughts:

>Does it mean that "Expenses waiting on you" will be something, that by default gives you "outstanding" and "draft" status. 

Yes - the default searches basically just pre-populate filters

>What should be the behaviour of default searches? Can we choose the "Expenses waiting on you" and then filter it even more with category filter(for example).

Going to lean on Shawn, Jason, and Carlos here, but I think yes I think you can add additional filters, _but_ once you add another filter parameter, the LHN item would no longer look "selected". I'm thinking of these default searches as relevant starting points which are helpful out of the box but you can then manipulate further if you so choose. Does that sound right to everyone?

**Comment by luacmartins** [comment]
Yea, I think the behavior doesn't really change. All we're doing it adding a new "canned search" with a different set of filters applied, in this case `status:outstanding,drafts`

**Comment by trjExpensify** [comment]
What's the intention for "Expenses waiting on you" though? Is it:

- Any `open` report that is waiting for you to submit
- Any `processing` report , that you need to approve. 
- Any `approved` report, that you need to pay. 

Are the results grouped by report as well, or just a pure transaction list?

**Comment by SzymczakJ** [comment]
FYI I'm OOO tomorrow, I'll try to put up a draft PR at the beginning of next week.

**Comment by trjExpensify** [comment]
>I would think it just uses this query: type:expense status:drafts,outstanding to:user@domain.com

Hm, are draft reports `to:` yourself until you submit them though? 🤔 

>Which means we'd see expenses grouped by reports. And yeah, I think that means open reports waiting to be submitted, as well as processing/approved reports.

If we're including approved reports, we need to add the `approved` status into the query then yeah?

**Comment by shawnborton** [comment]
Yeah, not entirely sure - do you have a good suggestion for what the query should be? In general I agree that the intention is [this](https://github.com/Expensify/App/issues/52317#issuecomment-2536720330), I'm just not sure how we best achieve that with the query.

**Comment by SzymczakJ** [comment]
Chill Melvin, working on it.

**Comment by trjExpensify** [comment]
>Yeah, not entirely sure - do you have a good suggestion for what the query should be? In general I agree that the intention is https://github.com/Expensify/App/issues/52317#issuecomment-2536720330, I'm just not sure how we best achieve that with the query.

Yeah, I think it's pretty hard to put them all together basically because you'll have a "from" and "to" mix, and when you do that, you end up with results you don't want (i.e as a submitter, all of your outstanding (`processing`) reports). It makes me think that we should have two suggestions really. So one being "Unsubmitted expenses". The other being expenses you need to approve or pay if you're an approver/reimburser. 

"Draft expenses" - `type:expense status:draft from:submitter@domain.com group-by:report`
"Expenses waiting on you" - `type:expense status:outstanding, approved to:approver@domain.com group-by:report`

Maybe even call that second one something more on the nose with "Expenses to approve/pay".

**Comment by trjExpensify** [comment]
>Btw, I don't think Expenses waiting on you would appear unless you are a submitsTo or an approvesTo in the configured workspace.

Ah, interesting.. the workspace's designated reimburser might not be either of those. How do we account for that for expense reports waiting to be paid? 🤔

**Comment by JmillsExpensify** [comment]
For that one, we could use the workspace's designated reimburser, just like you mentioned. So you get the `Expenses waiting on you` suggestion if you are any of:

- `submitsTo`
- `forwardsTo`
- Designated reimburser

**Comment by trjExpensify** [comment]
Hm, so this one won't return any results for the designated reimburser unless they are the also the `submitsTo` or `forwardsTo` though? 

>"Expenses waiting on you" - type:expense status:outstanding, approved to:[signed-in-user]@domain.com

To show reports waiting to be paid to that user when "direct" reimbursement is enabled, their query is more accurately `type:expense status:approved, closed` with no "to" distinction. Two things come to mind given that: 

- Should we use a different suggestion for "Expenses to pay" and seed that to the designated reimburser on the workspace, or all admins if the workspace settings has reimbursementChoice `manual` ("Indirect")?
- Should we also consider including something in the query to ensure that the reports that show up here have at least one expense on the report that's reimbursable? Otherwise a report with all card transactions on it will show.

**Comment by SzymczakJ** [comment]
Update:
I decided to split this issue into two parts: 
- 1st part will add status filter, put search types in the place of status, and add new canned queries
- 2nd part will add SearchInput to mobile view of SearchPage, update SearchPage mobile view to match designs, update SearchPage so we always show SearchInput and improve code quality in Search module

**PR for 1st part should be ready for swm review tomorrow**
Is it okay with you if I split this that way @luacmartins? If we don't split this the PR might grow to enormous size

**Comment by luacmartins** [comment]
Yea, that's totally fine.

**Comment by SzymczakJ** [comment]
So @JmillsExpensify @trjExpensify what should we put in canned searches? For now I added All expenses(`type:expense status:all`) and Expenses waiting for you(`type:expense status:outstanding,approved to:[signed-in-user]@domain.com`), but I suppose that's not the full list.
I will create one more PR for this issue, in which we could update it, so take your time discussing it, but I'm just bumping the discussion.

**Comment by SzymczakJ** [comment]
What would be the icons for these @dannymcclain ?

**Comment by shawnborton** [comment]
Hmm maybe something like this? cc @Expensify/design for thoughts though... 
![CleanShot 2024-12-18 at 12 03 41@2x](https://github.com/user-attachments/assets/1358409a-a6e9-4488-a735-d21c9dad9e4f)

**Comment by dannymcclain** [comment]
Yeah I like those Shawn. If we wanted to go W I L D we could add small hourglasses to the thumbs up and money bag icons like we have with the credit card icon:
![CleanShot 2024-12-18 at 11 07 27@2x](https://github.com/user-attachments/assets/256c6389-4a6e-48f0-b423-ddba73450a2d)

But I'm not sure that's super necessary.

**Comment by shawnborton** [comment]
Oh I would be down to see something like that if you feel like exploring it! I don't feel strongly, but I like the idea.

**Comment by dannymcclain** [comment]
I did a little bit of exploring [here](https://www.figma.com/design/X3wpgP8YdhiVbtp9XNdlLJ/Icons?node-id=208-3675&t=oSA92YcJpdsLMHwI-11), not a huge fan TBH haha.

**Comment by shawnborton** [comment]
Hah yeah, I can see that. Kinda feels slightly forced maybe. So maybe we just go with the simple ones for now?

**Comment by dannymcclain** [comment]
>So maybe we just go with the simple ones for now?

Yeah I think this is the right move. So in summary, let's go with what Shawn posted [here](https://github.com/Expensify/App/issues/52317#issuecomment-2551848580).

**Comment by JmillsExpensify** [comment]
Coming back to this issue, are we sure that we should do this right now? I'm nervous about implementing "suggested" searches before we've published an extensive design doc, especially with the HL "Search v3" design on the horizon. 

Additionally, this plan with `all`, `draft`, `approve`, `pay` is highly dependent on workspace set up. Most workspaces would have the first three, though there is a enough variability even based on use case (employee, approver, admin), that I think we should honestly pump the breaks on this initiative and wait for "Search v3" to work itself through the design process. 

I'm sorry, I know that when we first created this issue, things were different. So this goes back on why we started this issue in the first place. But the planning dynamics have changed quite a bit since then, so I think it's best that we put a HOLD back on this issue, probably until the more specific design doc for "suggested" searches goes out. 

Anyone else feeling similar?

**Comment by SzymczakJ** [comment]
This task consists of two independent parts:
1. Adding new suggested searches, which were under our discussion.
2. Always showing search input on desktop and mobile views, which doesn't touch any of the suggested searches logic and can be implemented independently.
Unfortunately I've already created PR for part 1 😭, but I think we could put that part 1 PR on hold and I could start working on part 2. Or do you think we should put both parts on hold because the whole design will change @JmillsExpensify @trjExpensify @shawnborton

**Comment by trjExpensify** [comment]
>Anyone else feeling similar?

Agreed.

>but I think we could put that part 1 PR on hold and I could start working on part 2

Yep, exactly. Let's focus on the search input here for now, and then we can look to take the PR to move the type tabs and replace them with suggestions off hold later.

**Comment by JmillsExpensify** [comment]
Ok cool. I like working on part 2 now. That's effectively just a superficial "design" change that is consistent with the future.

**Comment by SzymczakJ** [comment]
Then I'll start it right away.

**Comment by shawnborton** [comment]
So I think that gives us something like this then, right? 

![image](https://github.com/user-attachments/assets/ca7ac5dc-1f01-410f-841e-2fc26fb3c334)

**Comment by shawnborton** [comment]
And for mobile, I think this is what we want in the meantime as well:
![CleanShot 2024-12-19 at 11 08 44@2x](https://github.com/user-attachments/assets/6ce8c881-92f8-4c5f-b424-87859a2e90e6)

**Comment by JmillsExpensify** [comment]
@shawnborton one small comment on the second mock. Wouldn't the saved searches use the bookmark icon? Or are we going to change that out for the magnifying glass. This is what saved searches look like in production. 

![CleanShot 2024-12-20 at 11 08 16@2x](https://github.com/user-attachments/assets/2da6d610-4c03-429e-8ce0-ed5f346fdc66)

**Comment by shawnborton** [comment]
Ah yeah, that's my bad. Updated mock here: 
![CleanShot 2024-12-20 at 08 51 52@2x](https://github.com/user-attachments/assets/e145091d-b6cb-4fe1-b151-13322ad7d4f6)

**Comment by shawnborton** [comment]
Also note that we want to update the search input to use a height of 40px, like so: 
![CleanShot 2024-12-20 at 08 52 28@2x](https://github.com/user-attachments/assets/6366d1e4-ad29-43e1-8b29-be8119fe0ef1)

**Comment by dannymcclain** [comment]
>Also note that we want to update the search input to use a height of 40px, like so:

But just on mobile!

**Comment by dannymcclain** [comment]
OH! Yeah ok I just did a little doodle in Figma to see what it's like on desktop and I'm good with it!

**Comment by SzymczakJ** [comment]
I'm working on the PR I linked above, it should enter review process this week.

**Comment by SzymczakJ** [comment]
Update: The refactor posed more problems that I initially thought, but I'm almost done with it.
Along with this refactor we will try to optimize Search Page, as it becomes slower and slower, so it may take us a day/two more

**Comment by SzymczakJ** [comment]
As it turns out the Reports Page performance was a blocker for us. I suspect that adding SearchRouterInput to it(with the RouterList underneath) was just a critical point when the whole page became just too heavy.
I will start working on performance issue right away and try to open up two PRs: 
1. overalll performance boost for Reports Page, 
2. reopen the old PR which I will also review and make more performant than it was before

WDYT about this plan @luacmartins?

**Comment by luacmartins** [comment]
Sounds good! Thanks for working on this!

**Comment by ikevin127** [comment]
@SzymczakJ https://github.com/Expensify/App/issues/52317#issuecomment-2624841124 sounds good to me!

We had a report of somebody encountering a `Maximum update depth exceded.` error when opening Reports page, which neither me nor **luacmartins** were able to reproduce while testing the PR, here's the Slack [conversation](https://expensify.slack.com/archives/C01GTK53T8Q/p1738170941379309), not sure there's much to do about it since we don't know why it's happening on their side but I thought I should mention it since a rework is planned.

**Comment by SzymczakJ** [comment]
Update: 
after some investigation, we have a promising solution for part of SearchPage performance problem, which implementation is still in progress. I will have to measure it thoroughly, so I'm sure it doesn't break anything else and I will post more detailed description when I'm sure it works.
@Kicu is also working on this issue, which is really helpful!

**Comment by luacmartins** [comment]
Thanks @SzymczakJ. That sounds promising.

**Comment by SzymczakJ** [comment]
Update:
1. I've tweaked the performance of my PR, so it doesn't add any additional overhead, when we compare it to main(according to the profiling that I've made, on average we're 20 ms faster on the navigation to Reports Page). Is it possible to give build from this PR to the testing team to make sure that on real devices the performance is not affected @luacmartins ?
2. Unfortunately I've noticed that on main, performance of navigating to Reports Page is much worse than it was last week(at least on simulators). I wasn't able to find the PR that slowed down the performance, but I wanted to let you know. I'll look for it root cause of it tomorrow.
3. I have a lot of ideas on how to speed up Search Page more and I would like to continue work on this, but it would be great to merge this PR first, since it doesn't intoduce performance issues anymore(of course after it being tested by test team).
WDYT @luacmartins

**Comment by luacmartins** [comment]
>Is it possible to give build from this PR to the testing team to make sure that on real devices the performance is not affected @luacmartins ?

I kicked off a build and will ask QA to test it once ready.

>Unfortunately I've noticed that on main, performance of navigating to Reports Page is much worse than it was last week(at least on simulators). I wasn't able to find the PR that slowed down the performance, but I wanted to let you know. I'll look for it root cause of it tomorrow.

That's unfortunate, but not surprising given the number of changes we're making to Search. The only PR I'm aware of that was merged recently is [this one](https://github.com/Expensify/App/pull/50135)

>I have a lot of ideas on how to speed up Search Page more and I would like to continue work on this, but it would be great to merge this PR first, since it doesn't intoduce performance issues anymore(of course after it being tested by test team).
WDYT @luacmartins 

I agree. I think we should make incremental changes here and tackle this with smaller PRs.

**Comment by garrettmknight** [comment]
@anmurali can you update the payment summary [here](https://github.com/Expensify/App/issues/52317#issuecomment-2617538695)?

**Comment by luacmartins** [comment]
@SzymczakJ we merged the PR, but we still have to address the issues below:

- https://github.com/Expensify/App/pull/56326#issuecomment-2666309609
- https://github.com/Expensify/App/pull/56326#issuecomment-2666842774

**Comment by shawnborton** [comment]
Let's also make sure we close the router on desktop when the user presses the escape key, cc @luacmartins to add that to your follow up list.

**Comment by garrettmknight** [comment]
@anmurali bump on the payment summary, plz.

**Comment by ikevin127** [comment]
> * [x] [@ikevin127] Please propose [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps to ensure the new feature will work correctly on production in further releases.

Regarding ☝ **BZ Checklist**, I think no regression test is needed for https://github.com/Expensify/App/pull/56326 as it's a UI refactor.

**Comment by lschurr** [comment]
Looks like @anmurali is already assigned from BZ, so I'm going to remove my assignment.

**Comment by garrettmknight** [comment]
@lschurr Anu's no longer on BZ.

**Comment by lschurr** [comment]
Payment summary:

- Contributor on the [first PR](https://github.com/Expensify/App/pull/55343): @Kicu (no payment needed)
- Reviewer: $125 @rayane-d (to be paid in NewDot)
- Contributor on the [second PR](https://github.com/Expensify/App/pull/56326): @SzymczakJ (no payment needed)
- Reviewer: $250 @ikevin127 (paid in [Upwork](https://www.upwork.com/nx/wm/workroom/39806658/overview))

**Comment by lschurr** [comment]
We discussed internally and realized there was an error in payment on this one.

The correct payment summary is as follows:

- Contributor: @Kicu (no payment needed)
- Reviewer: $125 @rayane-d (to be paid in NewDot)
- Contributor: @SzymczakJ (no payment needed)
- Reviewer: $125 @ikevin127

The explanation is here:

- [Simplify single line TextInput styling to allow for different heights](https://github.com/Expensify/App/pull/55343) reviewed by @rayane-d, which caused [this regression](https://github.com/Expensify/App/issues/55662)
- [Redesign search to always show search input(mobile + desktop)](https://github.com/Expensify/App/pull/55078) reviewed by @ikevin127, which caused a few regressions, e.g. [this one](https://github.com/Expensify/App/issues/56032)

Due to the error, I've requested a refund of $125 in Upwork from @ikevin127

**Comment by ikevin127** [comment]
@lschurr I think there's a misunderstanding here due to multiple PRs and different contributors being paid in the same issue.

Here's a summary of my work for this issue:
1. Reviewed first PR: [
Redesign search to always show search input(mobile + desktop)](https://github.com/Expensify/App/pull/55078) as C+, which had a few regressions and was fully [reverted](https://github.com/Expensify/App/pull/56053) - meaning no payment for me on this one.
2. Reviewed the second PR: [Add search input to mobile search page(with fixed performance)](https://github.com/Expensify/App/pull/56326) that was a rework of the first one, which did not cause any regressions as far as I know - we merged with 2 UI issues which CME agreed on having follow-up PRs for.

Since we started from scratch on the second PR rework, and I performed another complete review (including past regressions testing) - it was correct to get the full $250 payment.

cc @luacmartins for confirmation

##

If confirmed otherwise, I noted down the refund request and will send it as soon as I get some available balance.

**Comment by luacmartins** [comment]
>Reviewed first PR:
https://github.com/Expensify/App/pull/55078 as C+, which had a few regressions and was fully https://github.com/Expensify/App/pull/56053 - meaning no payment for me on this one.
Reviewed the second PR: https://github.com/Expensify/App/pull/56326 that was a rework of the first one, which did not cause any regressions as far as I know - we merged with 2 UI issues which CME agreed on having follow-up PRs for.

@ikevin127 AFAIK this counts as a regression whether we fully revert the PR or work on a new fix for it.

**Comment by garrettmknight** [comment]
Oops, typo - approved $125.
